hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateMap.java

```diff
@@ -87,8 +78,7 @@ void addPipeline(Pipeline pipeline) throws IOException {
     }
     pipeline2container.put(pipeline.getId(), new TreeSet<>());
     if (pipeline.getPipelineState() == PipelineState.OPEN) {
-      query2OpenPipelines.computeIfAbsent(pipeline.getReplicationConfig(),
-          any -> new CopyOnWriteArrayList<>()).add(pipeline);
+      query2OpenPipelines.computeIfAbsent(pipeline.getReplicationConfig(), any -> new ArrayList<>()).add(pipeline);
```

Comment:
nit: I think this layout is more readable:

```suggestion
      query2OpenPipelines.computeIfAbsent(pipeline.getReplicationConfig(), any -> new ArrayList<>())
          .add(pipeline);
```